src/BigQuery/Connection/Rest.php

```diff
@@ -0,0 +1,258 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\BigQuery\Connection;
+
+use Google\Cloud\BigQuery\Connection\ConnectionInterface;
+use Google\Cloud\RequestBuilder;
+use Google\Cloud\RequestWrapper;
+use Google\Cloud\RestTrait;
+use Google\Cloud\Upload\AbstractUploader;
+use Google\Cloud\Upload\MultipartUploader;
+use Google\Cloud\Upload\ResumableUploader;
+use Google\Cloud\UriTrait;
+use GuzzleHttp\Psr7;
+use GuzzleHttp\Psr7\Request;
+
+/**
+ * Implementation of the
+ * [Google BigQuery JSON API](https://cloud.google.com/bigquery/docs/reference/v2/).
+ */
+class Rest implements ConnectionInterface
+{
+    use RestTrait;
+    use UriTrait;
+
+    const BASE_URI = 'https://www.googleapis.com/bigquery/v2/';
+    const UPLOAD_URI = 'https://www.googleapis.com/upload/bigquery/v2/projects/{projectId}/jobs';
+
+    /**
+     * @param array $config
+     */
+    public function __construct(array $config = [])
+    {
+        $this->setRequestWrapper(new RequestWrapper($config));
+        $this->setRequestBuilder(new RequestBuilder(
+            __DIR__ . '/ServiceDefinition/bigquery-v2.json',
```

Comment:
We could script this as part of our CI job. If the json schema falls out of sync, we'll fail the build and prompt to update the schema in the codebase.